java/integTest/src/test/java/glide/BaseClientTests.java

```diff
@@ -0,0 +1,416 @@
+    }
+
+    @Test
+    @SneakyThrows
+    public void incr_commands_existing_key() {
+        String key = RandomStringUtils.randomAlphabetic(10);
+
+        for (BaseClient client : clients) {
+            assertEquals(OK, client.set(key, "10").get(10, SECONDS));
+
+            assertEquals(client.incr(key).get(10, SECONDS), 11);
+            assertEquals(client.get(key).get(10, SECONDS), "11");
+
+            assertEquals(client.incrBy(key, 4).get(10, SECONDS), 15);
+            assertEquals(client.get(key).get(10, SECONDS), "15");
+
+            assertEquals(client.incrByFloat(key, 5.5).get(10, SECONDS), 20.5);
+            assertEquals(client.get(key).get(10, SECONDS), "20.5");
+        }
+    }
+
+    @Test
+    @SneakyThrows
+    public void incr_commands_non_existing_key() {
+        String key1 = RandomStringUtils.randomAlphabetic(10);
+        String key2 = RandomStringUtils.randomAlphabetic(10);
+        String key3 = RandomStringUtils.randomAlphabetic(10);
+
+        for (BaseClient client : clients) {
+            assertNull(client.get(key1).get(10, SECONDS));
+            assertEquals(client.incr(key1).get(10, SECONDS), 1);
+            assertEquals(client.get(key1).get(10, SECONDS), "1");
+
+            assertNull(client.get(key2).get(10, SECONDS));
+            assertEquals(client.incrBy(key2, 3).get(10, SECONDS), 3);
+            assertEquals(client.get(key2).get(10, SECONDS), "3");
+
+            assertNull(client.get(key3).get(10, SECONDS));
+            assertEquals(client.incrByFloat(key3, 0.5).get(10, SECONDS), 0.5);
+            assertEquals(client.get(key3).get(10, SECONDS), "0.5");
+        }
+    }
+
+    @Test
+    @SneakyThrows
+    public void test_incr_commands_with_str_value() {
+        String key1 = RandomStringUtils.randomAlphabetic(10);
+
+        for (BaseClient client : clients) {
+            assertEquals(OK, client.set(key1, "foo").get(10, SECONDS));
+            Exception incrException =
+                    assertThrows(ExecutionException.class, () -> client.incr(key1).get(10, SECONDS));
+
+            assertTrue(incrException.getCause() instanceof RequestException);
+            assertTrue(incrException.getCause().getMessage().contains("value is not an integer"));
+
+            Exception incrByException =
+                    assertThrows(ExecutionException.class, () -> client.incrBy(key1, 3).get(10, SECONDS));
+
+            assertTrue(incrByException.getCause() instanceof RequestException);
+            assertTrue(incrByException.getCause().getMessage().contains("value is not an integer"));
+
+            Exception incrByFloatException =
+                    assertThrows(
+                            ExecutionException.class, () -> client.incrByFloat(key1, 3.5).get(10, SECONDS));
+
+            assertTrue(incrByFloatException.getCause() instanceof RequestException);
+            assertTrue(
+                    incrByFloatException.getCause().getMessage().contains("value is not a valid float"));
```

Comment:
test `incr*` commands with missing key too
Consider testing with `Long.MAX_VALUE` too (and max double)

Reply:
Are you referring to null key value or not existing in the DB? I can add the maxes I copied over all the IT tests from python and node.

Reply:
Try both - null and missing.
Re: max value - you will get an error on Redis side, rust core lib or in java. Good to know and record the limitations.

Reply:
Fair but does that mean we need to write out the same test for all functions that incr decr by a number? There multiple variations for hash_commands etc...

Reply:
Well, yes. If we have the same error for null, missing and max values - you can combine those into a single paramtrized test.